lib/transformers/typescript.js

```diff
@@ -5,9 +5,45 @@ const {
   getTsJestConfig,
   stripInlineSourceMap,
   getCustomTransformer,
-  getVueJestConfig
+  getVueJestConfig,
+  warn
 } = require('../utils')
 
+function getDiagnostics(tsBlob, options, ts) {
+  ensureRequire('tempy', ['tempy'])
+  const tempy = require('tempy')
+
+  const temp = tempy.writeSync(tsBlob, { extension: 'ts' })
+  let program = ts.createProgram([temp], options)
+  program.emit()
+
+  let allDiagnostics = ts
+    .getPreEmitDiagnostics(program)
```

Comment:
`getPreEmitDiagnostics` is usually slow. Probably you can try to use [this](https://github.com/kulshekhar/ts-jest/blob/452639242d7f59502bda7aa94b81735f9c9ad203/src/compiler/language-service.ts#L21)